lms/static/scripts/frontend_apps/components/test/JSTORPicker-test.js

```diff
@@ -164,57 +128,24 @@ describe('JSTORPicker', () => {
       );
     });
 
-    it('validates entered URL when input is focused and "Enter" is pressed', () => {
-      const wrapper = renderJSTORPicker();
-      const input = getInput(wrapper);
-      const keyEvent = new Event('keydown');
-      keyEvent.key = 'Enter';
-
-      setURL(wrapper, 'https://www.jstor.org/stable/1234');
-
-      input.getDOMNode().dispatchEvent(keyEvent);
-
-      assert.calledOnce(fakeArticleIdFromUserInput);
-      assert.calledWith(
-        fakeArticleIdFromUserInput,
-        'https://www.jstor.org/stable/1234'
-      );
-    });
-
-    it('confirms the validated URL if "Enter" is pressed subsequently', () => {
+    it('confirms the validated URL if select button is clicked', () => {
       const onSelectURL = sinon.stub();
-      const wrapper = renderJSTORPicker({ onSelectURL });
-      const input = getInput(wrapper);
-      const keyEvent = new Event('keydown');
-      keyEvent.key = 'Enter';
-
-      setURL(wrapper, 'https://www.jstor.org/stable/1234');
-
-      // First enter press will validate URL
-      interact(wrapper, () => {
-        input.getDOMNode().dispatchEvent(keyEvent);
+      const wrapper = renderJSTORPicker({
+        onSelectURL,
+        defaultArticle: '1234',
       });
-      assert.calledOnce(fakeArticleIdFromUserInput);
 
       simulateMetadataFetch(wrapper, 'test title');
 
-      // Enter will submit if terms are checked and there is valid metadata fetched
-      input.getDOMNode().dispatchEvent(keyEvent);
+      wrapper
+        .find('Button[data-testid="select-button"]')
+        .find('button')
+        .simulate('click');
 
       assert.calledOnce(onSelectURL);
       assert.calledWith(onSelectURL, 'jstor://1234');
     });
 
-    it('validates entered URL when `IconButton` is clicked', () => {
-      const wrapper = renderJSTORPicker();
-      setURL(wrapper, 'foo');
-
-      wrapper.find('IconButton button[title="Find article"]').simulate('click');
-
-      assert.calledOnce(fakeArticleIdFromUserInput);
-      assert.calledWith(fakeArticleIdFromUserInput, 'foo');
-    });
```

Comment:
The logic when the IconButton is pressed is covered in `URLFormWithPreview-test.js`